env0/resource_environment.go

```diff
@@ -255,18 +263,23 @@ func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta
 			return diagErr
 		}
 
+		createTemplatePayload.IsSingleUse = true
+
 		name := d.Get("name").(string)
 		createTemplatePayload.Name = "single-use-template-for-" + name
 
-		template, err := apiClient.TemplateCreate(createTemplatePayload)
+		template, err = apiClient.TemplateCreate(createTemplatePayload)
 		if err != nil {
 			return diag.Errorf("could not create template: %v", err)
 		}
 
-		d.Set("template.0.id", template.Id)
+		values := d.Get("template")
```

Comment:
This ugly... but the previous line is invalid...
This is the only way to do these type of Sets.